### ISSUE TITLE
BigDecimal has bad allocate+initialize sequence

### ISSUE THREAD

**Issue by headius** [jruby/jruby#365]
See #198 for details.

Basically, our BigDecimal defines "new" instead of an allocator and initialize method. Initialize is never called, and until my fix for #198, we did not observe subclasses at all. We need to correct this to allow subclassing BigDecimal.

**Comment by kares** [comment]
it is actually how 1.8.7 / 1.9.3 behaves with BigDecimal sub-classes (does not invoke `initialize`) 
... but the bug has been open for long enough to be ripe for 9K since 2.2.2 does call `initialize` :)

**Comment by enebo** [comment]
To give this a little more traction here is a test extracted from @kares test for 1.8/1.9 (but I inverted it):

```ruby
require 'bigdecimal'

class BigDeci < BigDecimal
  attr_reader :called

    # MRI does not invoke initialize on 1.8./1.9
    def initialize(arg); @called = true; super(arg.to_s) end

    def abs; -super end
    def infinite?; false end

end


a = BigDeci.new 1.to_s
p a.called #=> true
```

For anyone who wants to work on this the newInstance "new" needs to end up doing some of it in an "initialize".

**Comment by headius** [comment]
@guyboertje Yeah this is a problem because in standard Ruby, there's no BigDecimal class until you require it. In JRuby, it's still possible to create RubyBigDecimal instances before it has been initialized in the runtime.

This might be a good argument for having RubyBigDecimal lazily set itself up?
